akka-stream/src/main/scala/akka/stream/impl/io/FilePublisher.scala

```diff
@@ -52,6 +52,7 @@ private[akka] final class FilePublisher(f: Path, completionPromise: Promise[IORe
       chan = FileChannel.open(f, FilePublisher.Read)
     } catch {
       case ex: Exception ⇒
+        completionPromise.success(IOResult(0L, Failure(ex)))
```

Comment:
@patriknw, that will only be executed if the channel is not null, so there is zero risk for a collision.
@viktorklang, I agree, but that's out of scope for this PR.